### ISSUE TITLE
`autocmd BufRead * autocmd FileType <buffer> ++once` doesn't work

### ISSUE THREAD

**Issue by cryptomilk** [neovim/neovim#21920]
### Describe the bug

The autocmd described in `:help restore-cursor` doesn't work.

### Steps to reproduce

A simple reproducer would be:

```
autocmd BufRead * autocmd FileType <buffer> ++once echo "wurst"
```

### Expected behavior

It should print "wurst" once.

### Neovim version (nvim -v)

0.8.2

### Vim (not Nvim) behaves the same?

yes

### Operating system/version

Fedora 37

### Terminal name/version

konsole

### $TERM environment variable

tmux-256color

### Installation

system package manager

**Comment by cryptomilk** [comment]
I'm not sure if this is just a documentation issue.

The `autocmd BufRead *` set up an event and if it triggers, it wants to call `autocmd FileType <buffer> ++once echo "wurst"`. This looks totally valid to me.

**Comment by zeertzjq** [comment]
This works if you run it before `filetype.lua` is sourced, e.g, put it in `init.vim` or `init.lua` or pass it in `--cmd`, but not if you run it after startup because `filetype.lua` has already been loaded.

**Comment by cryptomilk** [comment]
Ah, it needs to be `BufReadPre`

**Comment by clason** [comment]
PR welcome! (We have already diverged from upstream for this snippet.)

**Comment by lewis6991** [comment]
To clarify, the fix should be to run filetype.lua as the last `BufRead` event.

**Comment by zeertzjq** [comment]
> To clarify, the fix should be to run filetype.lua as the last `BufRead` event.

How is that possible? By hardcoding that in C?

**Comment by lewis6991** [comment]
It might not be possible without C changes. Otherwise, we should close this as "won't fix".

**Comment by clason** [comment]
> It might not be possible without C changes. Otherwise, we should close this as "won't fix".

No, we also could (and should) simply update the snippet in the docs -- that's the PR I had in mind.
